src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,73 @@
+package mate.academy.springboot.datajpa.controller;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.springboot.datajpa.dto.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.ProductResponseDto;
+import mate.academy.springboot.datajpa.dto.mapper.DtoMapper;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final DtoMapper<Product, ProductRequestDto, ProductResponseDto> dtoMapper;
+
+    public ProductController(ProductService productService,
+                             DtoMapper<Product, ProductRequestDto, ProductResponseDto> dtoMapper) {
+        this.productService = productService;
+        this.dtoMapper = dtoMapper;
+    }
+
+    @PostMapping
+    public ProductResponseDto create(@RequestBody ProductRequestDto dto) {
+        return dtoMapper.mapToDto(productService.create(dtoMapper.mapToModel(dto)));
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return dtoMapper.mapToDto(productService.getById(id));
+    }
+
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id, @RequestBody ProductRequestDto dto) {
+        Product product = dtoMapper.mapToModel(dto);
+        product.setId(id);
+        return dtoMapper.mapToDto(productService.update(product));
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteById(@PathVariable Long id) {
+        productService.deleteById(id);
+    }
+
+    @GetMapping("/by-price")
+    public List<ProductResponseDto> getProductsByPriceBetween(
+            @RequestParam("from") BigDecimal fromPrice,
+            @RequestParam("to") BigDecimal toPrice) {
```

Comment:
Just rename params in from and to and you will not need write "from" or "to" inside RequestParam annotations.

Reply:
I don't want to change the naming of the input parameters because they are informative. Instead, I want to specify to write only "from" and "to" in the URL, because there is already a by-price there